lbfgs.go

```diff
@@ -1,52 +1,61 @@
+// Copyright ©2014 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
 package optimize
 
 import (
 	"github.com/gonum/floats"
 )
 
-// LBFGS implements the limited-memory BFGS algorithm. While the normal BFGS algorithm
-// makes a full approximation to the inverse hessian, LBFGS instead approximates the
-// hessian from the last Store optimization steps. The Store parameter is a tradeoff
-// between cost of the method and accuracy of the hessian approximation.
-// LBFGS has a cost (both in memory and time) of O(Store * inputDimension).
-// Since BFGS has a cost of O(inputDimension^2), LBFGS is more appropriate
-// for very large problems. This "forgetful" nature of LBFGS may also make it perform
-// better than BFGS for functions with Hessians that vary rapidly spatially.
+// LBFGS implements the limited-memory BFGS optimization method.
+//
+// While the normal BFGS algorithm makes a full approximation to the inverse
+// Hessian, LBFGS instead approximates the Hessian from the last Store
+// iterations. The Store parameter is a tradeoff between cost of the method and
+// accuracy of the Hessian approximation. LBFGS has a cost (both in memory and
+// time) of O(Store * dimension).
 //
-// If Store is 0, Store is defaulted to 15.
-// A Linesearcher for LBFGS must satisfy the strong Wolfe conditions at every
-// iteration. If Linesearcher == nil, an appropriate default is chosen.
+// Since BFGS has a cost of O(dimension^2), LBFGS is more appropriate for very
```

Comment:
I would flip the sentence

LBFGS is more appropriate than BFGS for large dimensional problems as the cost of LBFGS scales as O(Store * dim) while BFGS scales with O(dim^2).